pkg/networkpath/traceroute/tcp/tcpv4.go

```diff
@@ -86,6 +82,15 @@ func (t *TCPv4) TracerouteSequential() (*Results, error) {
 		return nil, fmt.Errorf("failed to get raw ICMP listener: %w", err)
 	}
 
+	// Create a TCP listener with port 0 to get a random port from the OS
+	// and reserve it for the duration of the traceroute
+	port, tcpListener, err := reserveLocalPort()
```

Comment:
It might be preferable to use create a `SOCK_STREAM` socket yourself and call `bind()` (but without calling `listen()`) which will just reserve the TCP port